cpu/qn908x/include/vendor/QN908XC.h

```diff
@@ -0,0 +1,9484 @@
+/*
+** ###################################################################
+**     Processors:          QN9080C
+**                          QN9083C
+**
+**     Compilers:           Keil ARM C/C++ Compiler
+**                          GNU C Compiler
+**                          IAR ANSI C/C++ Compiler for ARM
+**                          MCUXpresso Compiler
+**
+**     Reference manual:    QN908X User manual Rev.1.0 21 Mar 2017
+**     Version:             rev. 1.0, 2017-03-27
+**     Build:               b170424
+**
+**     Abstract:
+**         CMSIS Peripheral Access Layer for QN908XC
+**
+**     Copyright 1997-2016 Freescale Semiconductor, Inc.
+**     Copyright 2016-2017 NXP
+**     All rights reserved.
+**
+**     SPDX-License-Identifier: BSD-3-Clause
+**
+**     http:                 www.nxp.com
+**     mail:                 support@nxp.com
+**
+**     Revisions:
+**     - rev. 1.0 (2017-03-27)
+**         Initial version.
+**
+** ###################################################################
+*/
+
+/*!
+ * @file QN908XC.h
+ * @version 1.0
+ * @date 2017-03-27
+ * @brief CMSIS Peripheral Access Layer for QN908XC
+ *
+ * CMSIS Peripheral Access Layer for QN908XC
+ */
+
+#ifndef _QN908XC_H_
+#define _QN908XC_H_                              /**< Symbol preventing repeated inclusion */
+
```

Comment:
I hate modifying vendor files just as much as the next guy, but those need a

```C
#ifdef __cplusplus
extern "C" {
#endif

…

#ifdef __cplusplus
}
#endif
```

header guard.
[CI is complaining](https://github.com/RIOT-OS/RIOT/pull/13855/checks?check_run_id=1483161312) - usually vendor files are excluded from CI checks, but I see the point since they might transitively end up getting included in a global header file.

It's only 5 files, so I guess this is the easiest way.

Reply:
Well, the thing is that the files that need this have it around the functions already. If you only have #define or typedef you don't need this extern "C" header guard, which is the case for those files. I can add them but it would be just to calm down externc/check.sh. I can deal with providing some tooling to update the vendor files that patch this in when needed. Vendor SDK updates are not that common anyway.

Reply:
Vendor files also use "#if defined(__cplusplus)" but the check expects "#ifdef __cplusplus"

Reply:
> Vendor files also use "#if defined(__cplusplus)" but the check expects "#ifdef __cplusplus"

urgh, yea `dist/tools/externc/check.sh` doesn't check for this - but this is a bug in the script.
If you could fix that, that would be great!
Not sure if there is a good way to determine if a header file only contains defines.